server/ai_http.go

```diff
@@ -96,26 +97,95 @@ func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 
 		remoteAddr := getRemoteAddr(r)
 		ctx := clog.AddVal(r.Context(), clog.ClientIP, remoteAddr)
+		requestID := string(core.RandomManifestID())
 
-		// skipping handleAIRequest for now until we have payments
-
-		var (
-			mid        = string(core.RandomManifestID())
-			pubUrl     = TrickleHTTPPath + mid
-			subUrl     = pubUrl + "-out"
-			controlUrl = pubUrl + "-control"
-		)
-		jsonData, err := json.Marshal(
-			&worker.LiveVideoToVideoResponse{
-				PublishUrl:   pubUrl,
-				SubscribeUrl: subUrl,
-				ControlUrl:   controlUrl,
-			})
+		payment, err := getPayment(r.Header.Get(paymentHeader))
 		if err != nil {
+			respondWithError(w, err.Error(), http.StatusPaymentRequired)
+			return
+		}
+		sender := getPaymentSender(payment)
+
+		var req worker.GenLiveVideoToVideoJSONRequestBody
+		if err := jsonDecoder[worker.GenLiveVideoToVideoJSONRequestBody](&req, r); err != nil {
+			respondWithError(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		orch := h.orchestrator
+		pipeline := "live-video-to-video"
+		cap := core.Capability_LiveVideoToVideo
+		modelID := *req.ModelId
+		clog.V(common.VERBOSE).Infof(ctx, "Received request id=%v cap=%v modelID=%v", requestID, cap, modelID)
+
+		// Create storage for the request (for AI Workers, must run before CheckAICapacity)
+		err = orch.CreateStorageForRequest(requestID)
+		if err != nil {
+			respondWithError(w, "Could not create storage to receive results", http.StatusInternalServerError)
+		}
+
+		// Check if there is capacity for the request
+		if !orch.CheckAICapacity(pipeline, modelID) {
+			respondWithError(w, fmt.Sprintf("Insufficient capacity for pipeline=%v modelID=%v", pipeline, modelID), http.StatusServiceUnavailable)
+			return
+		}
+
+		// Start trickle server for live-video
+		mid := string(core.RandomManifestID())
+		pubUrl := TrickleHTTPPath + mid
+		subUrl := pubUrl + "-out"
+
+		//Append orchestrator service url host to the urls
+		host := h.node.GetServiceURI().String()
+		publishUrl, _ := common.AppendHostname(pubUrl, host)
+		req.PublishUrl = publishUrl.String()
+
+		subscribeUrl, _ := common.AppendHostname(subUrl, host)
+		req.SubscribeUrl = subscribeUrl.String()
+
+		controlUrl, _ := common.AppendHostname(pubUrl+"-control", host)
+		controlUrlStr := controlUrl.String()
+		req.ControlUrl = &controlUrlStr
+
+		// Prepare request to worker
+		workerReq := worker.LiveVideoToVideoParams{
+			ModelId:      req.ModelId,
+			PublishUrl:   req.SubscribeUrl, // SubscribeUrl is the publish url for the worker
+			SubscribeUrl: req.PublishUrl,   // PublishUrl is the subscribe url for the worker
+			ControlUrl:   req.ControlUrl,
+			Params:       req.Params,
+		}
+
+		// append params to the request
+		queryParams := r.FormValue("query")
```

Comment:
I also don't understand, do we pass all `query` and `params` from Gateway into Orchestrator?

Reply:
I've fixed this now so that it uses the `Params` field from ai_mediaserver.go https://github.com/livepeer/go-livepeer/blob/527bd90fd0e053f9c49741fe124ff3f58c11fca6/server/ai_mediaserver.go#L489-L493

https://github.com/livepeer/go-livepeer/blob/527bd90fd0e053f9c49741fe124ff3f58c11fca6/server/ai_http.go#L163C1-L170C4